server/selective_sync_test.go

```diff
@@ -194,7 +194,7 @@ func TestChatMembersSpanPlatforms(t *testing.T) {
 
 	t.Run("empty set of channel members", func(t *testing.T) {
 		chatMembersSpanPlatforms, appErr := th.p.ChatMembersSpanPlatforms(model.ChannelMembers{})
-		require.Nil(t, appErr)
+		require.Error(t, appErr)
```

Comment:
I don't believe we can use `require.Error` with a function returning `*model.AppError`, because a `nil` value would still pass:
```go
package main_test

import (
	"testing"

	"github.com/mattermost/mattermost/server/public/model"
	"github.com/stretchr/testify/require"
)

func TestAppError(t *testing.T) {
	var nilAppErr *model.AppError
	require.Error(t, nilAppErr)
}
```

```
> go test .
ok      error   0.189s
```

Let's either use `require.NotNil` or better yet, change the code in `selective_sync.go` to use the `p.apiClient` instead of `p.API`, with the former managing this complexity and always returning a proper `error` interface that works with `require.Error`.